core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -507,12 +507,7 @@ protected Scope visitRefreshMaterializedView(RefreshMaterializedView refreshMate
                 throw semanticException(TABLE_NOT_FOUND, refreshMaterializedView, "Table '%s' does not exist", targetTable);
             }
 
-            if (targetTableHandle.isPresent() && metadata.getMaterializedViewFreshness(session, name).isMaterializedViewFresh()) {
-                analysis.setSkipMaterializedViewRefresh(true);
-            }
-            else {
-                analysis.setSkipMaterializedViewRefresh(false);
-            }
+            analysis.setSkipMaterializedViewRefresh(metadata.getMaterializedViewFreshness(session, name).isMaterializedViewFresh());
```

Comment:
Why `targetTableHandle.isPresent()` is being removed?

Reply:
It is already checked two lines above